### PR TITLE
Allow Skipping Benchmark Errors

### PR DESCRIPTION
**Pull request by shawntabrizi**
This PR introduces yet another `BenchmarkError` type which can allow the benchmarking pipeline to skip a benchmark when it returns an error, but continue the rest of the benchmarking pipeline. Not really great for pallet benchmarks, but can be helpful when skipping XCM features which are not enabled on the chain.

### CODE REVIEW

File: frame/benchmarking/src/lib.rs [hunk 1]
```diff
@@ -1220,6 +1220,11 @@ macro_rules! impl_benchmark_test_suite {
 									},
 									$crate::BenchmarkError::Override(_) => {
 										// This is still considered a success condition.
+										$crate::log::error!("WARNING: benchmark error overrided");
```
Comment (emostov) [inline frame/benchmarking/src/lib.rs:1223]: Would it be helpful to still log `benchmark_name` with the logs in `Override` and `Skip`? Might bake it more clear whats going on

**Review by emostov — APPROVED** [review]

File: frame/benchmarking/src/lib.rs (hunk 1 above)
Comment (apopiak) [reply frame/benchmarking/src/lib.rs:1223]: Why log with `error!` if it's a warning?

**Review by apopiak — APPROVED** [review]

File: frame/benchmarking/src/lib.rs (hunk 1 above)
Comment (shawntabrizi) [reply frame/benchmarking/src/lib.rs:1223]: @apopiak otherwise it would not appear in the output by default

**Comment by shawntabrizi** [comment]
bot merge

**Comment by shawntabrizi** [comment]
failing CI is a flaky test

**Comment by shawntabrizi** [comment]
bot force merge

**Comment by shawntabrizi** [comment]
bot merge force

**Comment by shawntabrizi** [comment]
bot merge force

**Comment by bkchr** [comment]
bot merge

**Comment by shawntabrizi** [comment]
bot merge

**Pull request merged**